icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -96,6 +167,111 @@ void dumpEnumeratedProperty(UProperty uproperty, FILE* f) {
     usrc_writeUCPTrie(f, shortPropName, utrie.getAlias(), UPRV_TARGET_SYNTAX_TOML);
 }
 
+void dumpScriptExtensions(FILE* f) {
+    IcuToolErrorCode status("icuexportdata: dumpScriptExtensions");
+
+    const char* scxFullPropName = u_getPropertyName(UCHAR_SCRIPT_EXTENSIONS, U_LONG_PROPERTY_NAME);
+    const char* scxShortPropName = u_getPropertyName(UCHAR_SCRIPT_EXTENSIONS, U_SHORT_PROPERTY_NAME);
+
+    // We want to use 16-bits for our exported trie of sc/scx data because we
+    // need 12 bits to match the 12 bits of data stored for sc/scx in the trie
+    // in the uprops.icu data file.
+    UCPTrieValueWidth scWidth = UCPTRIE_VALUE_BITS_16;
+
+    // Create a mutable UCPTrie builder populated with Script property values data.
+    const UCPMap* scInvMap = u_getIntPropertyMap(UCHAR_SCRIPT, status);
+    handleError(status, scxFullPropName);
+    LocalUMutableCPTriePointer builder(umutablecptrie_fromUCPMap(scInvMap, status));
+    handleError(status, scxFullPropName);
+
+    fputs("[[script_extensions]]\n", f);
+    fprintf(f, "long_name = \"%s\"\n", scxFullPropName);
+    if (scxShortPropName) fprintf(f, "short_name = \"%s\"\n", scxShortPropName);
```

Comment:
Minor -- I would move the printing of the names next to the fetching of the names. Those two really go together. Otherwise one of the names just happens to be used in handleError().